.github/workflows/auto-cherry-pick.yml

```
@@ -0,0 +1,263 @@
+                  repo: context.repo.repo,
+                  pull_number: context.payload.issue.number
+                });
+                
+                if (pr.merged && pr.base.ref === 'develop') {
+                  commitSha = pr.merge_commit_sha;
+                  targetBranches = pickMatch[1].split(',').map(b => b.trim());
+                  prNumber = context.payload.issue.number;
+                  prTitle = pr.title;
+                } else {
+                  shouldRun = false;
+                  core.setFailed('PR must be merged and target develop branch');
+                }
+              }
+            }
+
+            core.setOutput('should-run', shouldRun);
+            core.setOutput('commit-sha', commitSha);
+            core.setOutput('target-branches', JSON.stringify(targetBranches));
+            core.setOutput('pr-number', prNumber);
+            core.setOutput('pr-title', prTitle);
+
+            console.log(`Should run: ${shouldRun}`);
+            console.log(`Commit SHA: ${commitSha}`);
+            console.log(`Target branches: ${JSON.stringify(targetBranches)}`);
+
+  cherry-pick:
+    needs: validate-trigger
+    if: needs.validate-trigger.outputs.should-run == 'true'
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        target-branch: ${{ fromJson(needs.validate-trigger.outputs.target-branches) }}
+      fail-fast: false
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Configure Git
+        run: |
+          git config --global user.name "github-actions[bot]"
+          git config --global user.email "github-actions[bot]@users.noreply.github.com"
+
+      - name: Set up GitHub CLI
+        run: |
+          echo "${{ secrets.GITHUB_TOKEN }}" | gh auth login --with-token
+
+      - name: Validate target branch exists
+        run: |
+          if ! git ls-remote --exit-code --heads origin "${{ matrix.target-branch }}"; then
+            echo "❌ Target branch '${{ matrix.target-branch }}' does not exist"
+            exit 1
+          fi
+          echo "✅ Target branch '${{ matrix.target-branch }}' exists"
+
+      - name: Execute cherry-pick
+        id: cherry-pick
+        run: |
+          COMMIT_SHA="${{ needs.validate-trigger.outputs.commit-sha }}"
+          TARGET_BRANCH="${{ matrix.target-branch }}"
+          SHORT_SHA="${COMMIT_SHA:0:7}"
+          BRANCH_NAME="cherry-pick/${SHORT_SHA}-to-${TARGET_BRANCH}"
+          PR_NUMBER="${{ needs.validate-trigger.outputs.pr-number }}"
+          PR_TITLE="${{ needs.validate-trigger.outputs.pr-title }}"
+
+          echo "commit-sha=${COMMIT_SHA}" >> $GITHUB_OUTPUT
+          echo "target-branch=${TARGET_BRANCH}" >> $GITHUB_OUTPUT
+          echo "short-sha=${SHORT_SHA}" >> $GITHUB_OUTPUT
+          echo "branch-name=${BRANCH_NAME}" >> $GITHUB_OUTPUT
+          echo "pr-number=${PR_NUMBER}" >> $GITHUB_OUTPUT
```

```suggestion
          echo "commit-sha=${COMMIT_SHA}" | tee -a "$GITHUB_OUTPUT"
          echo "target-branch=${TARGET_BRANCH}" | tee -a "$GITHUB_OUTPUT"
          echo "short-sha=${SHORT_SHA}" | tee -a "$GITHUB_OUTPUT"
          echo "branch-name=${BRANCH_NAME}" | tee -a "$GITHUB_OUTPUT"
          echo "pr-number=${PR_NUMBER}" | tee -a "$GITHUB_OUTPUT"
```

Total nit, but if you pipe to `tee`, we can see the output in the logs (handy for debugging).